app/models/match.rb

```diff
@@ -25,4 +25,50 @@ def matches
   belongs_to :team_home, polymorphic: true
   belongs_to :team_away, polymorphic: true
   belongs_to :event, dependent: :delete
+
+  def name
+    winner_team = winner
+    if winner_team != nil
+      return winner_team.name
+    end
+    round
+  end
+
+  def depth
+    event.max_match_level - gameday
+  end
+
+  def round
+    key = { 0 => 'zero', 1 => 'one', 2 => 'two', 3 => 'three' }[depth]
+    key ||= 'other'
+    I18n.t('matches.round_name.' + key, round: (gameday + 1).to_s, gameid: index.to_s)
+  end
+
+  def has_winner?
+    points_home != nil && points_away != nil && points_home != points_away
+  end
+
+  def winner
+    if has_winner?
+      points_home > points_away ? team_home_recursive : team_away_recursive
```

Comment:
Does having more points always make you the winner?
In Golf, having a low score is better.
Maybe create a new method for this, so it can be changed more easily later

Reply:
yes